packages/matchbox/src/helpers/event.ts

```diff
@@ -0,0 +1,35 @@
+export function debounce<T extends (...args: unknown[]) => unknown>(
+  func: T,
+  wait: number,
+  immediate: boolean,
+) {
+  let timeout;
+  return function(...rest) { // eslint-disable-line
+    const context = this; // eslint-disable-line
+    const args = rest;
+    const later = function () {
+      timeout = null;
+      if (!immediate) {
+        func.apply(context, args);
+      }
+    };
+    const callNow = immediate && !timeout;
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+    if (callNow) {
+      func.apply(context, args);
+    }
+  };
+}
+
+export function noop(): void {
+  return undefined;
+}
+
+export function identity<T>(a: T) {
+  return a;
+}
+
+export function isNotTouchEvent(e: TouchEvent): boolean {
```

Comment:
Not sure if the React version of this type is compatible/required here, but the type used by `onTouchStart` is `onTouchStart?: React.TouchEventHandler<HTMLDivElement>`

Reply:
Maybe `isNotTouchEvent<T extends HTMLElement>(e: React.TouchEventHandler<T>)`

Up to you though, looks like the build passes anyway

Reply:
Oh wait i think i gave you bad advice, this is the even object not the event handler